activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/MessageProducerActivitiEventListener.java

```diff
@@ -24,10 +29,34 @@ public MessageProducerActivitiEventListener(ProcessEngineChannels producer,
 
     @Override
     public void onEvent(ActivitiEvent event) {
+        CommandContext currentCommandContext = Context.getCommandContext();
         ProcessEngineEvent newEvent = converterContext.from(event);
-        if (newEvent != null) {
-            producer.auditProducer().send(MessageBuilder.withPayload(newEvent).build());
+        if (newEvent == null) {
+            return;
+        }
+        @SuppressWarnings("unchecked")
+        List<ProcessEngineEvent> events = (List<ProcessEngineEvent>) currentCommandContext
+                                                                                          .getAttribute(MessageProducerCommandContextCloseListener.PROCESS_ENGINE_EVENTS);
+        if (events != null) {
+            events.add(newEvent);
+        } else {
+            events = new ArrayList<ProcessEngineEvent>();
```

Comment:
Thank you for your advice. I fixed.
https://github.com/Activiti/Activiti/pull/1424/commits/5d74bf002f8b8537147b9b0096feb4bc415b659f